### PR TITLE
Implement JSON DataFormat

### PR DESCRIPTION
**Pull request by jamierocks**
[API](https://github.com/SpongePowered/SpongeAPI/pull/1363) | **Common**

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [submodule "SpongeAPI"]
 	path = SpongeAPI
-	url = https://github.com/SpongePowered/SpongeAPI.git
+	url = https://github.com/jamierocks/SpongeAPI.git
```
Comment (kashike) [inline .gitmodules]: Don't include this in your primary commit.

Comment (jamierocks) [reply .gitmodules]: Resolved

File: src/main/java/org/spongepowered/common/data/persistence/JsonDataFormat.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+public class JsonDataFormat extends SpongeCatalogType implements DataFormat {
+
+    public JsonDataFormat(String id) {
+        super(id);
+    }
+
+    @Override
+    public DataContainer readFrom(InputStream input) throws InvalidDataFormatException, IOException {
+        GsonConfigurationLoader loader = GsonConfigurationLoader.builder()
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/data/persistence/JsonDataFormat.java:51]: Gson vs. Jackson.
With configurate, `configurate-json` is Jackson, while `configurate-gson` is Gson.

Comment (jamierocks) [reply src/main/java/org/spongepowered/common/data/persistence/JsonDataFormat.java:51]: pff jackson :p

**Comment by stephan-gh** [comment]
See https://github.com/SpongePowered/SpongeCommon/pull/1299

**Pull request closed without merging**